### PR TITLE
add use all points to the extract edges filter

### PR DESCRIPTION
**Pull request by akaszynski**
Enable the usage of ``SetUseAllPoints``. This improves the performance of the `extract_all_edges` at the potential expense increased memory usage.

### CODE REVIEW

File: tests/test_filters.py [hunk 1]
```diff
@@ -502,11 +502,18 @@ def test_extract_geometry_extent(uniform):
...
+def test_extract_all_edges(datasets):
     for dataset in datasets:
-        wire = dataset.extract_all_edges(progress_bar=True)
-        assert wire is not None
-        assert isinstance(wire, pyvista.PolyData)
+        edges = dataset.extract_all_edges()
+        assert edges is not None
+        assert isinstance(edges, pyvista.PolyData)
+
+    if pyvista.vtk_version_info < (9,):
```
Comment (akaszynski) [inline tests/test_filters.py:511]:
```suggestion
    if pyvista.vtk_version_info < (9, 1):
```

**Review by tkoyama010 — DISMISSED** [review]
LGTM.

File: pyvista/core/filters/data_set.py [hunk 2]
```diff
@@ -1339,6 +1348,11 @@ def extract_all_edges(self, progress_bar=False):
         """
         alg = _vtk.vtkExtractEdges()
         alg.SetInputDataObject(self)
+        if use_all_points:
+            try:
+                alg.SetUseAllPoints(use_all_points)
+            except AttributeError:  # pragma: no cover
+                raise VTKVersionError('This version of VTK does not support `use_all_points=True`.')
```
Comment (banesullivan) [inline pyvista/core/filters/data_set.py:1355]: Can we document which version this feature was introduced?

Comment (banesullivan) [reply pyvista/core/filters/data_set.py:1355]: From test, it looks like 9.1

**Review by banesullivan — APPROVED** [review]
LGTM, though minor feedback

File: pyvista/core/filters/data_set.py (hunk 2 above)
Comment (akaszynski) [reply pyvista/core/filters/data_set.py:1355]:
```suggestion
            except AttributeError:  # pragma: no cover
                raise VTKVersionError(
                    'This version of VTK does not support `use_all_points=True`. '
                    'VTK v9.1 or newer is required.'
                )
```

**Review by banesullivan — APPROVED** [review]

**Review by tkoyama010 — APPROVED** [review]

**Pull request merged**